### DISCUSSION TITLE
not adding 4g networks,but fiber connection is adding perfectly

### DISCUSSION THREAD

**Discussion by Bumpalert** [Ysurac/openmptcprouter#1758] [Q&A]
![experiment4_on4g](https://user-images.githubusercontent.com/56679716/115147019-b2b93780-a076-11eb-9ce6-df2560b00b8f.jpg)
![experiment4_usb1](https://user-images.githubusercontent.com/56679716/115147024-b6e55500-a076-11eb-8665-b2a5fa96711d.jpg)
![experiment5](https://user-images.githubusercontent.com/56679716/115147030-b8af1880-a076-11eb-9047-66793a8af545.jpg)
![experiment5_result_speed](https://user-images.githubusercontent.com/56679716/115147032-b9e04580-a076-11eb-9f7c-ccfa9ce9fa35.jpg)
![experiment5_usb0_act](https://user-images.githubusercontent.com/56679716/115147033-ba78dc00-a076-11eb-811c-4fb129809071.jpg)
![experiment5_usb1 ](https://user-images.githubusercontent.com/56679716/115147035-bb117280-a076-11eb-83f2-860bace67f2b.jpg)

**Comment by Bumpalert** [comment]
i am a newbie in openmptcprouter. I set up one in my RPI 4 & observed it has some issues when aggregating 4g connections from different vendors. It works perfectly when merging 2 connections from single vendor is merged. 

experiment 1: rate limit ACT fiber connection into 1 mbps or 5 mbps into 2 ports and connect to pi on ethernet to usb. It doubles the rate to 10mbps both in upload and download
experiment 2: act fiber at 5 mbps rate limited and same act connection through another router which is connecting to vpn in german ip, which adds a latency of 260ms(25ms is the typical delay of act connection). In this case the speed was shown 5mbps
experiment 3: keep one ACT fiber connection at 1mbps and add one Airtel 4G connection by usb tethering from android to pi. It adds traffic. act fiber is the master
experiment 4: remove fiber all together and add androdi usb tether vodafone 4g connection & one airtel 4g connection(one of them is master). this is not taking from usb1(as per realtime traffic) and all the traffic is from usb0(tried mptcp over vpn also ). But even though i set one of them master, it comes as mptcp enabled intead of master
experiment 5: use act fiber connection on wifi tethered over android usb for both usb1 & usb0  & disabled airtel 4g & vodafone 4g. This time I can see bandwidth is taken from both usb0 & usb1 equally.

**Comment by Bumpalert** [comment]
Sorry for making confusion for posting without questions.

My questions are 
1. Is the 4g vendors blocking mptcp? If yes, how to make tunnel? mptcp over vpn option also not working and my vpn usage is very less according to the statistics of vps.
2. how to find whether a vendor is blocking mptcp?
3. iperf is not giving any results for me. Does it need any special settings if I install vps script and pi installation correctly?

**Reply by Bumpalert** [reply]
@Ysurac please help

**Reply by Ysurac** [reply]
1. vendor can block MPTCP, you can try in Network->MPTCP, "MPTCP support check" tab each WAN. MPTCP over VPN should work, at least if UDP or VPN are not blocked.
2. see 1.
3. No, iperf should give results, but servers can be too much used. There is "omr-test-speed" on the router via SSH.

**Reply by Bumpalert** [reply]
Thanks for the quick response

Here goes the MPTCP response check response. I couldn't get any conclusion as result doesn't specifically say a pass or fail.


1: 192.168.42.129 
2: * 
3: * 
4: 122.15.225.134  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
5: * 
6: 99.82.183.36  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
7: * 
8: 52.95.64.190  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
9: * 
10: * 
11: 52.95.67.180  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
12: * 
13: * 
14: * 
15: * 
16: * 
17: 15.207.35.235 TCP::SrcPort TCP::DstPort TCP::SeqNumber TCP::AckNumber TCP::DataOffset TCP::Flags TCP::WindowsSize TCP::CheckSum IP::DiffServicesCP IP::TotalLength IP::Identification IP::TTL IP::CheckSum IP::SourceIP IP::DestinationIP +TCPOptionMaxSegSize TCPOptionMPTCPCapable::Sender's Key 

Result:
< Ethernet (14 bytes) :: DestinationMAC = e6:45:f5:8b:60:2f , SourceMAC = ce:fb:1d:bc:d0:9c , Type = 0x800 , >
< IP (20 bytes) :: Version = 4 , HeaderLength = 5 , DiffServicesCP = 10 , ExpCongestionNot = 0 , TotalLength = 56 , Identification = 0x0 , Flags = 2 , FragmentOffset = 0 , TTL = 49 , Protocol = 0x6 , CheckSum = 0x2ac4 , SourceIP = 15.207.35.235 , DestinationIP = 192.168.42.114 , >
< TCP (20 bytes) :: SrcPort = 65101 , DstPort = 3371 , SeqNumber = 743175713 , AckNumber = 92303596 , DataOffset = 9 , Reserved = 0 , Flags = ( SYN ACK ) , WindowsSize = 35764 , CheckSum = 0xe085 , UrgPointer = 0 , >
< TCPOptionMaxSegSize (4 bytes) :: Kind = 2 , Length = 4 , MaxSegSize = 1380 , >
< TCPOptionMPTCPCapable (12 bytes) :: Kind = 30 , Length = 12 , Subtype = 0 , Version = 0 , Checksum = 1 (Checksum Enabled) , Flags = 0 , Crypto = 1 (HMAC-SHA1) , Sender's Key = Sender's Key = 14559711581330927219 , >

**Reply by Ysurac** [reply]
You can see "TCPOptionMPTCPCapable::Sender's Key" at last hop, so it's not filtered.

**Comment by Bumpalert** [comment]
Observation 1:
got strange issue. wan1 has eth1.  wan2 has eth2. Both are 1mbps ports connected to same fiber connection. wan1 has an extra tenda router inbetween just to keep a different gateway address(to avoid a gateway conflict)

![image](https://user-images.githubusercontent.com/56679716/115155347-34bc5700-a09d-11eb-8132-e8ea9e4d3583.png)

wan1 is mptcp capable when checked on routerwebpage. But the eth1 is not passing in ssh

1: 192.168.2.1  [PARTIAL] +PartialTCP 
2: 192.168.13.1 IP::TTL IP::CheckSum 
3: 103.227.96.1  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
4: 10.248.5.12  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
5: 10.248.5.21  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
6: 10.248.5.33  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
7: 10.248.5.41  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
8: 99.83.64.100  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
9: 150.222.83.158 TCP::DataOffset TCP::CheckSum IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ] 
10: 52.93.79.167  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
11: * 
12: 150.222.243.164 TCP::DataOffset TCP::CheckSum IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ] 
13: * 
14: 52.95.66.144 TCP::DataOffset TCP::CheckSum IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ] 
15: 52.95.64.184  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
16: 52.95.64.183 TCP::DataOffset TCP::CheckSum IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ICMPExtensionMPLS ] 
17: 52.95.66.127 TCP::DataOffset TCP::CheckSum IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ] 
18: 52.95.67.180  [PARTIAL] +PartialTCP IP::TotalLength IP::TTL IP::CheckSum 
19: * 
20: * 
21: * 
22: * 
23: * 
24: 15.207.35.235 TCP::SrcPort TCP::DstPort TCP::SeqNumber TCP::AckNumber TCP::DataOffset TCP::Flags TCP::WindowsSize TCP::CheckSum IP::TotalLength IP::Identification IP::TTL IP::CheckSum IP::SourceIP IP::DestinationIP +TCPOptionMaxSegSize TCPOptionMPTCPCapable::Sender's Key 

Result:
< Ethernet (14 bytes) :: DestinationMAC = 00:e1:03:00:09:1a , SourceMAC = c8:3a:35:49:d1:b8 , Type = 0x800 , >
< IP (20 bytes) :: Version = 4 , HeaderLength = 5 , DiffServicesCP = 0 , ExpCongestionNot = 0 , TotalLength = 56 , Identification = 0x0 , Flags = 2 , FragmentOffset = 0 , TTL = 44 , Protocol = 0x6 , CheckSum = 0x585c , SourceIP = 15.207.35.235 , DestinationIP = 192.168.2.2 , >
< TCP (20 bytes) :: SrcPort = 65101 , DstPort = 51613 , SeqNumber = 2509320240 , AckNumber = 137308189 , DataOffset = 9 , Reserved = 0 , Flags = ( SYN ACK ) , WindowsSize = 35764 , CheckSum = 0xc13c , UrgPointer = 0 , >
< TCPOptionMaxSegSize (4 bytes) :: Kind = 2 , Length = 4 , MaxSegSize = 1452 , >
< TCPOptionMPTCPCapable (12 bytes) :: Kind = 30 , Length = 12 , Subtype = 0 , Version = 0 , Checksum = 1 (Checksum Enabled) , Flags = 0 , Crypto = 1 (HMAC-SHA1) , Sender's Key = Sender's Key = 12197871592008849876 , >



root@OpenMPTCProuter:~# omr-mptcp-intf eth1
Nay, Nay, Nay, your have an old computer that does not speak MPTCP. Shame on you!

But the speed test is actually doubling the speed


root@OpenMPTCProuter:~# omr-iperf
Connecting to host 15.207.35.235, port 65400
[  5] local 192.168.2.2 port 61567 connected to 15.207.35.235 port 65400
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec   632 KBytes  5.18 Mbits/sec    0   14.1 KBytes
[  5]   1.00-2.00   sec   293 KBytes  2.40 Mbits/sec    0   14.1 KBytes
[  5]   2.00-3.00   sec   266 KBytes  2.18 Mbits/sec    0   14.1 KBytes
[  5]   3.00-4.00   sec   133 KBytes  1.09 Mbits/sec    0   14.1 KBytes
[  5]   4.00-5.00   sec   275 KBytes  2.25 Mbits/sec    0   14.1 KBytes
[  5]   5.00-6.00   sec   269 KBytes  2.20 Mbits/sec    0   14.1 KBytes
[  5]   6.00-7.00   sec   270 KBytes  2.22 Mbits/sec    0   14.1 KBytes
[  5]   7.00-8.00   sec   137 KBytes  1.12 Mbits/sec    0   14.1 KBytes
[  5]   8.00-9.00   sec   269 KBytes  2.20 Mbits/sec    0   14.1 KBytes
[  5]   9.00-10.00  sec   273 KBytes  2.24 Mbits/sec    0   14.1 KBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  2.75 MBytes  2.31 Mbits/sec    0             sender
[  5]   0.00-10.08  sec  2.45 MBytes  2.04 Mbits/sec                  receiver

iperf Done.


when unplugged one port it comes 1mbps

root@OpenMPTCProuter:~# omr-iperf
Connecting to host 15.207.35.235, port 65400
[  5] local 192.168.2.2 port 61707 connected to 15.207.35.235 port 65400
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec   377 KBytes  3.09 Mbits/sec    0   14.1 KBytes
[  5]   1.00-2.00   sec  90.1 KBytes   738 Kbits/sec    0   14.1 KBytes
[  5]   2.00-3.00   sec   179 KBytes  1.47 Mbits/sec    0   14.1 KBytes
[  5]   3.00-4.00   sec  87.4 KBytes   715 Kbits/sec    0   14.1 KBytes
[  5]   4.00-5.00   sec  87.4 KBytes   716 Kbits/sec    0   14.1 KBytes
[  5]   5.00-6.00   sec   175 KBytes  1.43 Mbits/sec    0   14.1 KBytes
[  5]   6.00-7.00   sec  87.4 KBytes   715 Kbits/sec    0   14.1 KBytes
[  5]   7.00-8.00   sec   179 KBytes  1.47 Mbits/sec    0   14.1 KBytes
[  5]   8.00-9.00   sec  88.8 KBytes   727 Kbits/sec    0   14.1 KBytes
[  5]   9.00-10.00  sec  91.5 KBytes   750 Kbits/sec    0   14.1 KBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  1.41 MBytes  1.18 Mbits/sec    0             sender
[  5]   0.00-10.25  sec  1.24 MBytes  1.01 Mbits/sec                  receiver

iperf Done.


![image](https://user-images.githubusercontent.com/56679716/115155565-3a666c80-a09e-11eb-9fa1-6f4e6a78cc7c.png)



Observation 2:  omr-test-speed is not checking upload speed. any option to check upload speed as it is imported in video streaming application.

**Reply by Ysurac** [reply]
You know why I said to test via tracebox and not omr-mptcp-intf ;) This check is modified in 0.58beta.
You can check upload speed using iperf, nperf, speedtest,... omr-test-speed is only here to check if aggregation is working or not.

**Reply by Bumpalert** [reply]
Oh .. I am happy as omr-mptcp-intf is a known issue & quick responses...  I got another 4G operator who says a failure below as per my understanding.  Which vpn is recommended? Multi-link VPN (MLVPN) or Glorytun UDP (if any priority on performance)
Please confirm that this interface is not supporting mptcp as I am seeing a failure first time.


1: 192.168.42.129 
2: * 
3: 169.254.41.1 TCP::DataOffset TCP::CheckSum IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize 
4: 192.168.212.1 TCP::DataOffset TCP::CheckSum IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize +TCPOptionPad -TCPOptionMPTCPCapable 
5: 172.21.71.1  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
6: 122.15.177.142  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
7: 182.19.99.126  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
8: 99.82.183.36  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
9: 52.95.66.136 TCP::DataOffset TCP::CheckSum IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize +TCPOptionPad -TCPOptionMPTCPCapable [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ] 
10: 52.95.64.184  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
11: 52.95.64.191 TCP::DataOffset TCP::CheckSum IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize +TCPOptionPad -TCPOptionMPTCPCapable [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ICMPExtensionMPLS ] 
12: 52.95.66.215 TCP::DataOffset TCP::CheckSum IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum +TCPOptionMaxSegSize +TCPOptionPad -TCPOptionMPTCPCapable [Extra headers: ICMPExtension ICMPExtensionObject ICMPExtensionMPLS ] 
13: 52.95.65.133  [PARTIAL] +PartialTCP IP::DiffServicesCP IP::TotalLength IP::TTL IP::CheckSum 
14: * 
15: * 
16: * 
17: * 
18: * 
19: 15.207.35.235 TCP::SrcPort TCP::DstPort TCP::SeqNumber TCP::AckNumber TCP::DataOffset TCP::Flags TCP::WindowsSize TCP::CheckSum IP::DiffServicesCP IP::TotalLength IP::Identification IP::TTL IP::CheckSum IP::SourceIP IP::DestinationIP +TCPOptionMaxSegSize -TCPOptionMPTCPCapable 

Result:
< Ethernet (14 bytes) :: DestinationMAC = 4e:48:19:2f:79:1e , SourceMAC = ca:3c:38:c3:d4:37 , Type = 0x800 , >
< IP (20 bytes) :: Version = 4 , HeaderLength = 5 , DiffServicesCP = 10 , ExpCongestionNot = 0 , TotalLength = 44 , Identification = 0x0 , Flags = 2 , FragmentOffset = 0 , TTL = 47 , Protocol = 0x6 , CheckSum = 0x2d16 , SourceIP = 15.207.35.235 , DestinationIP = 192.168.42.44 , >
< TCP (20 bytes) :: SrcPort = 65101 , DstPort = 41097 , SeqNumber = 376823755 , AckNumber = 379934878 , DataOffset = 6 , Reserved = 0 , Flags = ( SYN ACK ) , WindowsSize = 35844 , CheckSum = 0xe977 , UrgPointer = 0 , >
< TCPOptionMaxSegSize (4 bytes) :: Kind = 2 , Length = 4 , MaxSegSize = 1380 , >

**Reply by Ysurac** [reply]
Yes it doesn't support MPTCP. You can use MPTCP over VPN. It will also work with Glorytun UDP or MLVPN

**Comment by Bumpalert** [comment]
hey 4g aggregated as per measurement lab, but ssh omr-test-speed is not improving which is confusing

![image](https://user-images.githubusercontent.com/56679716/115157713-5e2eb000-a0a8-11eb-9bc4-941e05217747.png)



Now turning on both interfaces,

![image](https://user-images.githubusercontent.com/56679716/115158062-fc6f4580-a0a9-11eb-9d97-43d0486ec852.png)


root@OpenMPTCProuter:~# omr-test-speed
Select best test server...
host: speedtest.frankfurt.linode.com - ping: 139
host: speedtest.tokyo2.linode.com - ping: 146
host: speedtest.singapore.linode.com - ping: 87
host: speedtest.newark.linode.com - ping: 224
host: speedtest.atlanta.linode.com - ping: 275
host: speedtest.dallas.linode.com - ping: 275
host: speedtest.fremont.linode.com - ping: 255
host: speedtest.tele2.net - ping: 139
host: www.ovh.net - ping:
host: speed.hetzner.de - ping: 140
host: ipv4.bouygues.testdebit.info - ping: 129
host: par.download.datapacket.com - ping: 208
host: nyc.download.datapacket.com - ping: 223
host: ams.download.datapacket.com - ping: 145
host: fra.download.datapacket.com - ping: 270
host: lon.download.datapacket.com - ping: 277
host: mad.download.datapacket.com - ping: 220
host: prg.download.datapacket.com - ping: 155
host: sto.download.datapacket.com - ping: 162
host: vie.download.datapacket.com - ping: 198
host: war.download.datapacket.com - ping: 159
host: atl.download.datapacket.com - ping: 249
host: chi.download.datapacket.com - ping: 266
host: lax.download.datapacket.com - ping: 255
host: mia.download.datapacket.com - ping: 251
host: nyc.download.datapacket.com - ping: 224
Best server is http://speedtest.singapore.linode.com/garbage.php?ckSize=10000, r                                                                                                                                                             unning test:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  154M    0  154M    0     0   592k      0 --:--:--  0:04:27 --:--:-- 1382k^C
root@OpenMPTCProuter:~#
root@OpenMPTCProuter:~#
root@OpenMPTCProuter:~# omr-test-speed usb0
Select best test server...
host: speedtest.frankfurt.linode.com - ping: 360
host: speedtest.tokyo2.linode.com - ping: 213
host: speedtest.singapore.linode.com - ping: 129
host: speedtest.newark.linode.com - ping: 305
host: speedtest.atlanta.linode.com - ping: 293
host: speedtest.dallas.linode.com - ping: 307
host: speedtest.fremont.linode.com - ping: 316
host: speedtest.tele2.net - ping: 215
host: www.ovh.net - ping:
host: speed.hetzner.de - ping: 356
host: ipv4.bouygues.testdebit.info - ping: 215
host: par.download.datapacket.com - ping: 612
host: nyc.download.datapacket.com - ping: 318
host: ams.download.datapacket.com - ping: 341
host: fra.download.datapacket.com - ping: 523
host: lon.download.datapacket.com - ping:
host: mad.download.datapacket.com - ping: 481
host: prg.download.datapacket.com - ping: 363
host: sto.download.datapacket.com - ping: 335
host: vie.download.datapacket.com - ping: 352
host: war.download.datapacket.com - ping: 237
host: atl.download.datapacket.com - ping: 448
host: chi.download.datapacket.com - ping: 500
host: lax.download.datapacket.com - ping: 451
host: mia.download.datapacket.com - ping: 314
host: nyc.download.datapacket.com - ping: 350
Best server is http://speedtest.singapore.linode.com/garbage.php?ckSize=10000, r                                                                                                                                                             unning test:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  107M    0  107M    0     0   834k      0 --:--:--  0:02:12 --:--:--  941k^C
root@OpenMPTCProuter:~# omr-test-speed eth2
Select best test server...
host: speedtest.frankfurt.linode.com - ping: 130
host: speedtest.tokyo2.linode.com - ping: 123
host: speedtest.singapore.linode.com - ping: 44
host: speedtest.newark.linode.com - ping: 224
host: speedtest.atlanta.linode.com - ping: 256
host: speedtest.dallas.linode.com - ping: 246
host: speedtest.fremont.linode.com - ping: 219
host: speedtest.tele2.net - ping: 126
host: www.ovh.net - ping:
host: speed.hetzner.de - ping: 135
host: ipv4.bouygues.testdebit.info - ping: 126
host: par.download.datapacket.com - ping: 175
host: nyc.download.datapacket.com - ping: 231
host: ams.download.datapacket.com - ping: 174
host: fra.download.datapacket.com - ping: 136
host: lon.download.datapacket.com - ping: 131
host: mad.download.datapacket.com - ping: 168
host: prg.download.datapacket.com - ping: 147
host: sto.download.datapacket.com - ping: 159
host: vie.download.datapacket.com - ping: 172
host: war.download.datapacket.com - ping: 159
host: atl.download.datapacket.com - ping: 241
host: chi.download.datapacket.com - ping: 244
host: lax.download.datapacket.com - ping: 216
host: mia.download.datapacket.com - ping: 246
host: nyc.download.datapacket.com - ping: 225
Best server is http://speedtest.singapore.linode.com/garbage.php?ckSize=10000, r                                                                                                                                                             unning test:
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 15.2M    0 15.2M    0     0   116k      0 --:--:--  0:02:13 --:--:--  263k^C


root@OpenMPTCProuter:~# omr-iperf
Connecting to host 15.207.35.235, port 65400
[  5] local 192.168.13.96 port 50129 connected to 15.207.35.235 port 65400
[ ID] Interval           Transfer     Bitrate         Retr  Cwnd
[  5]   0.00-1.00   sec   538 KBytes  4.40 Mbits/sec    0   14.1 KBytes
[  5]   1.00-2.00   sec   155 KBytes  1.27 Mbits/sec    0   14.1 KBytes
[  5]   2.00-3.00   sec   155 KBytes  1.27 Mbits/sec    0   14.1 KBytes
[  5]   3.00-4.00   sec  0.00 Bytes  0.00 bits/sec    0   14.1 KBytes
[  5]   4.00-5.00   sec   155 KBytes  1.27 Mbits/sec    0   14.1 KBytes
[  5]   5.00-6.00   sec   155 KBytes  1.27 Mbits/sec    0   14.1 KBytes
[  5]   6.00-7.00   sec   155 KBytes  1.27 Mbits/sec    0   14.1 KBytes
[  5]   7.00-8.00   sec   155 KBytes  1.27 Mbits/sec    0   14.1 KBytes
[  5]   8.00-9.00   sec  0.00 Bytes  0.00 bits/sec    0   14.1 KBytes
[  5]   9.00-10.00  sec   155 KBytes  1.27 Mbits/sec    0   14.1 KBytes
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bitrate         Retr
[  5]   0.00-10.00  sec  1.59 MBytes  1.33 Mbits/sec    0             sender
[  5]   0.00-10.07  sec  1.23 MBytes  1.03 Mbits/sec                  receiver

iperf Done.



![image](https://user-images.githubusercontent.com/56679716/115158039-e3669480-a0a9-11eb-9f0e-84ca9589f7a0.png)


2 confusions left
1. omr-iperf shows 1mbps always while measurement labs shows 3+Mbps when combined. measurement lab is consistent and if i remove one interfce it will show 1mbps. Is omr-iperf is taking only one path?
2. eth1/eth2 shows 111kbps which is  a 1mbps rate limited interface from fiber connection - very unusual behaviour- not a priority as not an actual use case. May be local traffic is 900kbps to keep other things running on the debug pc.

**Reply by Ysurac** [reply]
I see it use singapore server, connection can be limited depending of your location.

**Comment by Bumpalert** [comment]
1. omr-iperf is going to the vps right? It should be giving the added results,but that is not happening. As wireless speed varies, anyway to limit the wireless speed and validate the aggregation effect? Will SQM setting work? even though i told it works in previous post, i think the variable speed above 1mbps is tricking me. It was nice if i could limit 4G speed
2. Another problem is master setting not kept when done on the usb tether interface(it will be displayed as enabled in status screen), but setting is kept when set on eth1/2 interface. There are dns error reported in OMR. What is the function of Master/backup? Is it for DNS?
3. How to debug vpns in command line? Does it required to connect MPTCP supported carrier also over vpn if one of the carrier is not supporting mptcp? I am seeing a total unstability when I disconnect mptcp supported carrier(omr disconnects)
![image](https://user-images.githubusercontent.com/56679716/115186568-957c7b80-a0ff-11eb-86ed-b5c23013d13f.png)
